docs/modules/dind.md

```diff
@@ -0,0 +1,72 @@
+# DinD (Docker in Docker)
+
+Not available until the next release of testcontainers-go <a href="https://github.com/testcontainers/testcontainers-go"><span class="tc-version">:material-tag: main</span></a>
+
+## Introduction
+
+The Testcontainers module for DinD (Docker in Docker).
+
+## Adding this module to your project dependencies
+
+Please run the following command to add the DinD module to your Go dependencies:
+
+```
+go get github.com/testcontainers/testcontainers-go/modules/dind
+```
+
+## Usage example
+
+<!--codeinclude-->
+[Creating a DinD container](../../modules/dind/examples_test.go) inside_block:ExampleRun
+<!--/codeinclude-->
+
+## Module Reference
+
+### Run function
+
+- Not available until the next release of testcontainers-go <a href="https://github.com/testcontainers/testcontainers-go"><span class="tc-version">:material-tag: main</span></a>
+
+!!!info
+    The `RunContainer(ctx, opts...)` function is deprecated and will be removed in the next major release of _Testcontainers for Go_.
+
+The DinD module exposes one entrypoint function to create the DinD container, and this function receives three parameters:
+
+```golang
+func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustomizer) (*DinDContainer, error)
+```
+
+- `context.Context`, the Go context.
+- `string`, the Docker image to use.
+- `testcontainers.ContainerCustomizer`, a variadic argument for passing options.
+
+### Container Options
+
+When starting the DinD container, you can pass options in a variadic way to configure it.
+
+#### Image
+
+If you need to set a different DinD Docker image, you can set a valid Docker image as the second argument in the `Run` function.
+E.g. `Run(context.Background(), "/:")`.
+
+{% include "../features/common_functional_options.md" %}
+
+### Container Methods
+
+The DinD container exposes the following methods:
+
+#### Host
+
+The `Host` method returns the DinD URL, to be used for connecting
+to the Docker API using a Docker client. It'll be returned in the format of `string`.
+
+<!--codeinclude-->
+[Host](../../modules/dind/examples_test.go) inside_block:GetKubeConfig
```

Comment:
bug: I used a tag for the host snippet defined in a suggestion block below, and added a new tab for the getting client
```suggestion
[Host](../../modules/dind/examples_test.go) inside_block: didnHost
[Get a Docker client](../../modules/dind/examples_test.go) inside_block:getDockerClient
```